### PR TITLE
Add eslint-plugin-jest

### PR DESCRIPTION
**Pull request by bmish**
Add linter for tests: https://github.com/jest-community/eslint-plugin-jest

### CODE REVIEW

File: lib/__tests__/does-not-have-attribute.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ describe('assert.dom(...).doesNotHaveAttribute()', () => {
     ]);
   });
 
-  test('fails for wrong content (string value)', () => {
+  test('fails for wrong content (boolean value)', () => {
```
Comment (bmish) [inline lib/__tests__/does-not-have-attribute.ts:55]: Detected by [no-identical-title](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-identical-title.md).

File: lib/__tests__/is-visible.ts [hunk 2]
```diff
@@ -32,21 +32,6 @@ describe('assert.dom(...).isVisible()', () => {
         },
       ]);
     });
-
-    test('fails if element is missing', () => {
```
Comment (bmish) [inline lib/__tests__/is-visible.ts:36]: [no-identical-title](https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-identical-title.md), was a duplicate test case.

Comment (Turbo87) [reply lib/__tests__/is-visible.ts:36]: hmm, I think the original intention here was that the first test was the happy path with `assert.dom('h1').isVisible();` 😅

Comment (bmish) [reply lib/__tests__/is-visible.ts:36]: I see. I just checked and it looks like we can't test the happy path here due to:

```
 * JSDom based tests aren't able to discern visibility as we define it. Specifically,
 * the JSDom tests don't do layouting, therefore calculating `offsetWdith` or `offsetHeight`
 * won't work. As a result, we need to use Ember's test infrastructure to correctly assess
 * visibility, as those tests run in a browser environment.
 *
 * Additional tests cases of isVisible can be found in tests/acceptance/qunit-dom-test.js
```

So removing the test case should be fine.

Comment (Turbo87) [reply lib/__tests__/is-visible.ts:36]: makes sense, yeah

**Pull request merged**